app/src/main/java/com/bashkevich/androidfundamentals/movieslist/view/FragmentMoviesList.kt

```diff
@@ -37,30 +43,28 @@ class FragmentMoviesList : Fragment() {
 
         recyclerView = view.findViewById(R.id.movies_recycler_view)
 
-        recyclerView?.layoutManager = GridLayoutManager(view.context, 2)
-
-        recyclerView?.adapter = MoviesListAdapter(onMovieClickListener).apply {
-            scope.launch {
-                val movies = context?.let { loadMovies(it) }
-                if (movies != null) {
-                    bindMovies(movies)
-                }
-            }
+        moviesListViewModel.loadMoviesList()
+        moviesListViewModel.moviesListLiveData.observe(this.viewLifecycleOwner){movies->
+            setUpMoviesList(movies)
         }
+        setUpMoviesListRecyclerView()
```

Comment:
Might be late to setup recycler view here. Setup view should be before data binding.